### DISCUSSION TITLE
how do "do not download" certain file

### DISCUSSION THREAD

**Discussion by kucingkembar** [rmartin16/qbittorrent-api#192] [Q&A] [answered]
### What is the problem or limitation you are having?

hi, sorry for my bad English,
I like to set some of my download files to "do not download",
for example: a torrent file containing a tons combination of pdf and epub,
while I want to download all pdf from a torrent, but I want to "do not download" all epub files,
is this "software" capable to do that?
thank you

### Describe the solution you'd like

-----

### Describe alternatives you've considered

----

**Comment by rmartin16** [comment, marked as answer]
Yes; you can use [`torrents_file_priority()`](https://qbittorrent-api.readthedocs.io/en/latest/apidoc/torrents.html#qbittorrentapi.torrents.TorrentsAPIMixIn.torrents_file_priority).

For example:
```python
torrent = client.torrents_info(torrent_hashes="<torrent hash>")[0]
# show torrent files
print(torrent.files)
# get all file IDs for epub files
epub_file_ids = [file.id for file in torrent.files if file.name.endswith("epub")]
# set epub files to "do not download"
torrent.file_priority(file_ids=epub_file_ids, priority=0)
```

**Reply by kucingkembar** [reply]
thank you
